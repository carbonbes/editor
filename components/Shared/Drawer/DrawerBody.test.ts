import { describe, expect, test } from 'vitest'
import { DrawerBody } from './'
import mountDrawer from '~/tests/utils/mountDrawer'

describe('DrawerBody', () => {
  test('should be defined', () => {
    expect(DrawerBody).toBeDefined()
  })

  describe('should display', () => {
    function createDrawerBody(slot?: string) {
      return defineComponent({
        components: { DrawerBody },
        template: `
          <DrawerBody>
            ${slot ? slot : ''}
          </DrawerBody>
        `,
      })
    }

    test('correctly without the slot', async () => {
      const DrawerBodyComponent = createDrawerBody()
      const drawer = await mountDrawer(DrawerBodyComponent)
      const drawerBody = drawer.getComponent(DrawerBody)

      expect(drawerBody.html()).toMatchSnapshot()

      drawer.unmount()
    })

    test('correctly with the slot', async () => {
      const DrawerBodyComponent = createDrawerBody('<p>Test</p>')
      const drawer = await mountDrawer(DrawerBodyComponent)
      const drawerBody = drawer.getComponent(DrawerBody)

      expect(drawerBody.html()).toMatchSnapshot()

      drawer.unmount()
    })
  })
})
