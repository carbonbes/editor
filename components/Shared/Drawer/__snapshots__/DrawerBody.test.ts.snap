// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DrawerBody > should display > correctly with the slot 1`] = `
"<div data-dismissable-layer="" style="--snap-point-height: 0; pointer-events: auto;" id="" role="dialog" aria-describedby="" aria-labelledby="reka-dialog-title-v-0" data-state="open" data-vaul-drawer="" data-vaul-drawer-direction="bottom" data-vaul-delayed-snap-points="false" data-vaul-snap-points="false" tabindex="-1" class="fixed right-0 bottom-0 left-0 w-full h-full max-h-[95%] flex flex-col bg-white rounded-t-3xl"><span aria-hidden="true" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); clip-path: inset(50%); white-space: nowrap; word-wrap: normal;"><h2 id="reka-dialog-title-v-0"></h2></span>
  <p>Test</p>
</div>"
`;

exports[`DrawerBody > should display > correctly without the slot 1`] = `"<div data-dismissable-layer="" style="--snap-point-height: 0; pointer-events: auto;" id="" role="dialog" aria-describedby="" aria-labelledby="reka-dialog-title-v-0" data-state="open" data-vaul-drawer="" data-vaul-drawer-direction="bottom" data-vaul-delayed-snap-points="false" data-vaul-snap-points="false" tabindex="-1" class="fixed right-0 bottom-0 left-0 w-full h-full max-h-[95%] flex flex-col bg-white rounded-t-3xl"><span aria-hidden="true" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); clip-path: inset(50%); white-space: nowrap; word-wrap: normal;"><h2 id="reka-dialog-title-v-0"></h2></span></div>"`;
